posthog/api/feature_flag.py

```diff
@@ -121,9 +129,112 @@ def get_queryset(self) -> QuerySet:
+    _analytics_created_event_name = "feature flag override created"
+
+    def validate_override_value(self, value):
+        if not isinstance(value, str) and not isinstance(value, bool):
+            raise serializers.ValidationError(
+                f"Overridden feature flag value ('{value}') must be a string or a boolean.", code="invalid_feature_flag"
+            )
+        return value
+
+    def create(self, validated_data: Dict) -> FeatureFlagOverride:
+        feature_flag_override, created = FeatureFlagOverride.objects.update_or_create(
+            feature_flag=validated_data["feature_flag"],
+            user=validated_data["user"],
+            defaults={"override_value": validated_data["override_value"]},
+        )
+        request = self.context["request"]
+        if created:
+            posthoganalytics.capture(
+                request.user.distinct_id, self._analytics_created_event_name,
+            )
+        else:
+            posthoganalytics.capture(
+                request.user.distinct_id, self._analytics_updated_event_name,
+            )
+        return feature_flag_override
+
+    def update(self, instance: FeatureFlagOverride, validated_data: Dict) -> FeatureFlagOverride:
+        request = self.context["request"]
+        posthoganalytics.capture(
+            request.user.distinct_id, self._analytics_updated_event_name,
+        )
+        return super().update(instance, validated_data)
+
+
+class FeatureFlagOverrideViewset(
+    AnalyticsDestroyModelMixin, viewsets.ModelViewSet,
+):
+    queryset = FeatureFlagOverride.objects.all()
+    serializer_class = FeatureFlagOverrideSerializer
+    permission_classes = [IsAuthenticated, ProjectMembershipNecessaryPermissions]
+    authentication_classes = [
+        PersonalAPIKeyAuthentication,
+        TemporaryTokenAuthentication,
+        authentication.SessionAuthentication,
+        authentication.BasicAuthentication,
+    ]
+
+    def get_queryset(self) -> QuerySet:
+        if not self.request.user.is_authenticated:  # for mypy, since 'AnonymousUser' has no 'team'
+            raise exceptions.NotAuthenticated
+        queryset = (
+            super()
+            .get_queryset()
+            .filter(feature_flag__team=self.request.user.team, user__current_team=self.request.user.team)
+        )
+        if self.action == "list":
+            feature_flag_id = self.request.GET.get("feature_flag_id")
+            if feature_flag_id:
+                queryset = queryset.filter(feature_flag_id=feature_flag_id)
+        return queryset
+
+    def perform_create(self, serializer):
+        self._check_team_permissions(serializer)
+        return super().perform_create(serializer)
+
+    def perform_update(self, serializer):
+        self._check_team_permissions(serializer)
+        return super().perform_update(serializer)
+
+    def _check_team_permissions(self, serializer: FeatureFlagOverrideSerializer):
```

Comment:
I think this logic belongs on the serializer, in case it's called directly too and I would simplify it to only check if the user creating the override has permission (i.e. nested membership) to the FF & the user for which it's being created is part of the same org.